src/wacks4shop/shift4shop.py

```diff
@@ -156,9 +156,38 @@ def get_num_sales(self) -> int:
         num_sales = num_total_orders - num_not_completed_orders
         return num_sales
 
+    def get_num_sales(self, timestamp: str, prev_num_sales: int) -> int:
```

Comment:
is it possible to make `get_num_sales` and `get_legacy_num_sales` the same function, except one takes the extra args? and then conditionally use those args, instead of the duplicate logic? for example, default `prev_num_sales=0` and `timestamp=None` (so it becomes `Optional[str]`). I think in that case, you may not even have to update the internal code

Reply:
oh I guess you'd have to conditionally add the `limit` based on whether or not `datestart` exists. something like

```python
params = { "orderstatus": INCOMPLETE_ORDER_STATUS }
filters = {"datestart": timestamp} if timestamp else { "limit": 300 }
params.update(filters)
```

Reply:
addressed offline, but we want to keep two different signatures for these two functions. the plan is to extract common functionality into a helper function